Document HTTP controller method renaming to uppercase in XP 8.0 (#572)
enonic/doc-xp: docs/release/apps-upgrade.adoc

== API Deprecations and Breaking Changes

=== HTTP Controllers

HTTP controller method names have been changed to uppercase to avoid conflicts with JavaScript reserved words and imported library methods.

> The lowercase method names (`get`, `post`, `delete`, `put`, `patch`) are deprecated but still supported for backward compatibility. It is recommended to migrate to the new uppercase names (`GET`, `POST`, `DELETE`, `PUT`, `PATCH`). 

Reviewer: ```suggestion
The lowercase method names (`get`, `post`, `delete`, etc...) are deprecated but still supported for backward compatibility. It is recommended to migrate to the new uppercase names (`GET`, `POST`, `DELETE`, etc...). 
```

> NOTE: The `all` method remains lowercase and has not been changed to uppercase.

Reviewer: ```suggestion
NOTE: The `all` function remains lowercase and has not been changed to uppercase.
```

.XP 7 (deprecated but still works in XP 8)
[source,javascript]
----
exports.get = function(req) {
    return {
        body: 'Hello World',
        contentType: 'text/plain'
    };
};

exports.post = function(req) {
    // Handle POST
};

exports.delete = function(req) {
    // Handle DELETE
};

exports.all = function(req) {
    // Handle all other methods
};
----

.XP 8 (recommended)
[source,javascript]
----
exports.GET = function(req) {
    return {
        body: 'Hello World',
        contentType: 'text/plain'
    };
};

exports.POST = function(req) {
    // Handle POST
};

exports.DELETE = function(req) {
    // Handle DELETE
};

exports.all = function(req) {
    // Handle all other methods
};
----

The change was made to:

* Avoid conflicts with JavaScript reserved words like `delete`

> * Prevent naming conflicts when using ES6 module imports (e.g., `import {get} from '/lib/xp/content'`)

Reviewer: ```suggestion
```
